packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -5812,12 +5812,111 @@ void main() {
     state.bringIntoView(TextPosition(offset: controller.text.length));
 
     await tester.pumpAndSettle();
-    // The SingleChildScrollView is scrolled instead of the EditableText to
-    // reveal the caret.
+    // The SingleChildScrollView is scrolled instead of the EditableText to reveal the caret.
     expect(outerController.offset, outerController.position.maxScrollExtent);
     expect(editableScrollController.offset, 0);
   });
 
+  testWidgets('bringIntoView centers the viewport on caret when the caret is wider than the viewport', (WidgetTester tester) async {
+    const String text = 'to coz ze ze Szwecji';
+    final TextEditingController controller = TextEditingController(text: text);
+
+    await tester.pumpWidget(MaterialApp(
```

Comment:
I think that's the current behavior because `additoinalOffset` has the wrong sign

Reply:
That's correct: from the comments, what was intended in case of large rects was to center the viewport on the visible part of the rect. But due to the sign error, wrong offset is jumped to.

This bug makes it impossible to e.g.select by dragging where there are rects larger than the viewport (could be a very large font or a WidgetSpan in case of custom rich text editors) as instead of scrolling to te rect centre, the viewport will jump somewhere above the rect.

Reply:
Ah got it, I tried it again and it works to jump to the correct location.  Maybe I had switched off of this branch.  Sounds good.